### PR TITLE
Lcguang images by tag

### LINKED ISSUE #9: Images by Tag
#### As a user I want to view all images associated with a tag.

__Story__:

Now that we have added the ability for images to be associated with tags, let
us provide the ability for users to filter images by a given tag. Nana, who
loves fast planes, will be delighted to see only images labeled with SR-71.

__Acceptance criteria__:
- [ ] When I click on a tag, I see a filtered list of only the images that have
  the tag I clicked on.
- [ ] The tag name, not its ID, is used to perform the filtering.

__Discussion topics__:
- [ ] Where in the code should we implement this functionality?
- [ ] What happens if a user tries to manually filter by a nonexistent tag?

__Dependencies__:
- Image Tags

### PR DESCRIPTION
**Pull request by lcguang**
Closes #9

### CODE REVIEW

File: test/controllers/images_controller_test.rb [hunk 1]
```diff
@@ -1,6 +1,39 @@
...
+  test 'should show all images with nil tag' do
+    Image.create!(link: 'http://example.com/a.jpg', tag_list: 'tag_a')
+    Image.create!(link: 'http://example.com/b.jpg', tag_list: 'tag_b')
+    Image.create!(link: 'http://example.com/c.jpg', tag_list: 'tag_b')
+    get images_path, params: { tag: nil }
+    assert_response :ok
+    assert_select '.card', count: 3
+  end
+
+  test 'should show no image with tag that does not exit' do
```
Comment (Xin-Tong) [inline test/controllers/images_controller_test.rb]: It's great that you've thought about this test case.

File: app/controllers/images_controller.rb [hunk 2]
```diff
@@ -4,7 +4,11 @@ def new
   end
 
   def index
-    @images = Image.all.reverse
+    @images = if params[:tag].nil?
```
Comment (lcguang) [inline app/controllers/images_controller.rb]: It's because of the rubocop offenses...

File: app/controllers/images_controller.rb [hunk 3]
```diff
@@ -4,7 +4,11 @@ def new
...
+                Image.all.reverse
+              else
+                Image.tagged_with(params[:tag]).reverse
```
Comment (lcguang) [inline app/controllers/images_controller.rb]: Already done in the new PR.

File: test/controllers/images_controller_test.rb (hunk 1 above)
Comment (lcguang) [inline test/controllers/images_controller_test.rb]: Thanks. I was trying to find all the test cases.

**Pull request merged**